### ISSUE TITLE
would really like some help getting to run this on my local machine

### ISSUE THREAD

**Issue by vesper8** [KD1994/Body-Detection-With-BG-Removal#1]
Hi there! Thank you for sharing your project here. I would really like to give it a test drive but I'm struggling to getting it working

First I see all these imports at the top of your Server.py

```
from flask import Flask, request, render_template
from imageai.Detection import ObjectDetection
import requests
import os
from io import BytesIO
import numpy as np
from PIL import Image
import tensorflow as tf
import datetime
```

I already have some of these installed but am missing many. Why not have a requirements.txt file included to auto-install all these dependencies? That would be helpful!

Then I read your notes about adding the models to the models folder, however I've had a look at both the ImageAI folder and that other link you shared and I can't find the files in questions that I'm supposed to copy to the /models/ folder

What are the steps needed please? Do I have to clone those other two repos, build them and then the files will be generated?

I'm talking about
```
static/models/yolov3/yolo.h5
static/models/xception_model
```

I couldn't find those files inside 
https://github.com/OlafenwaMoses/ImageAI.git
and
https://github.com/tensorflow/models/blob/master/research/deeplab/deeplab_demo.ipynb

Sorry for the noobish questions.. but I would really like to get this to work as I need a reliable background remover for my current project.

Basically something of a step-by-step installation guide would be amazing!!

Thanks for the help!

**Comment by KD1994** [comment]
> Hi there! Thank you for sharing your project here. I would really like to give it a test drive but I'm struggling to getting it working
> 
> First I see all these imports at the top of your Server.py
> 
> ```
> from flask import Flask, request, render_template
> from imageai.Detection import ObjectDetection
> import requests
> import os
> from io import BytesIO
> import numpy as np
> from PIL import Image
> import tensorflow as tf
> import datetime
> ```
> I already have some of these installed but am missing many. Why not have a requirements.txt file included to auto-install all these dependencies? That would be helpful!
> 
> Then I read your notes about adding the models to the models folder, however I've had a look at both the ImageAI folder and that other link you shared and I can't find the files in questions that I'm supposed to copy to the /models/ folder
> 
> What are the steps needed please? Do I have to clone those other two repos, build them and then the files will be generated?
> 
> I'm talking about
> 
> ```
> static/models/yolov3/yolo.h5
> static/models/xception_model
> ```
> I couldn't find those files inside
> https://github.com/OlafenwaMoses/ImageAI.git
> and
> https://github.com/tensorflow/models/blob/master/research/deeplab/deeplab_demo.ipynb
> 
> Sorry for the noobish questions.. but I would really like to get this to work as I need a reliable background remover for my current project.
> 
> Basically something of a step-by-step installation guide would be amazing!!
> 
> Thanks for the help!

Hey, i'd really appreciate your suggestion and also pointing out the link issue.
let me update it and then you can enjoy it.

you don't have to clone other repos, you just need to download those 2 models and save it in /static/models/ the way I have done it.

**Comment by vesper8** [comment]
thanks a lot for updating your readme and clarifying everything!! seems a lot more approachable now! Will be giving it a shot very soon and let you know if I run into other troubles.

Out of curiosity.. how close would you say are your results in terms of quality compared to what https://remove.bg can delivery?

**Comment by KD1994** [comment]
I've been there and used it, the result is pretty much great, and for the things I have done you can check the output out in "static/op_images/" it's not bad but i wouldn't say it's awesome as you will get with remove.bg (That's my honest answer :-))!
